core/src/test/java/org/apache/calcite/test/SqlXmlFunctionsTest.java

```diff
@@ -36,6 +39,23 @@
  */
 class SqlXmlFunctionsTest {
 
+  private static final String XML = "<document>string</document>";
+  private static final String XSLT =
+      "<xsl:stylesheet xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\"></xsl:stylesheet>";
+  private static final String DOCUMENT_PATH = "/document";
+  private static @Nullable String xmlExternalEntity = null;
+  private static @Nullable String xsltExternalEntity = null;
+
+  @BeforeAll public static void setup() throws Exception {
+    final File testFile = File.createTempFile("foo", "temp");
```

Comment:
Although this is a temporary test file, the documentation for [java.io.File.createTempFile()](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#createTempFile(java.lang.String,%20java.lang.String)) notes that the [java.nio.Files.createTempFile()](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createTempFile-java.nio.file.Path-java.lang.String-java.lang.String-java.nio.file.attribute.FileAttribute...-) method creates the file with more restrictive permissions.

Reply:
Modification applied.
Let me know if everything is ok on your side @exceptionfactory . If so, I will proceed to finalize the PR (squash commits into a single one + merge).